hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/procedures/RunClusteringProcedure.scala

```diff
@@ -100,24 +111,59 @@ class RunClusteringProcedure extends BaseProcedure
         logInfo("No order columns")
     }
 
+    orderStrategy match {
+      case Some(o) =>
+        val strategy = LayoutOptimizationStrategy.fromValue(o.asInstanceOf[String])
+        conf = conf ++ Map(
+          HoodieClusteringConfig.LAYOUT_OPTIMIZE_STRATEGY.key() -> strategy.getValue
+        )
+      case _ =>
+        logInfo("No order strategy")
+    }
+
+    options match {
+      case Some(p) =>
+        val paramPairs = StringUtils.split(p.asInstanceOf[String], ",").asScala
+        paramPairs.foreach{ pair =>
+          val values = StringUtils.split(pair, "=")
+          conf = conf ++ Map(values.get(0) -> values.get(1))
+        }
+      case _ =>
+        logInfo("No options")
+    }
+
     // Get all pending clustering instants
     var pendingClustering = ClusteringUtils.getAllPendingClusteringPlans(metaClient)
       .iterator().asScala.map(_.getLeft.getTimestamp).toSeq.sortBy(f => f)
+
+    pendingClustering = instantsStr match {
+      case Some(inst) =>
+        operator = ClusteringOperator.EXECUTE
```

Comment:
here why we need set operator to EXECUTE but in line#144 we do not need?

Reply:
Because the user does not specify the instants

Reply:
I think we need check if users specify the instants with SCHEDULE and SCHEDULE_AND_EXECUTE, we should throw exception instead of set it to EXECUTE when specify  instants.

Reply:
please put the line `operator = ClusteringOperator.EXECUTE` below the line `logInfo("No op")` and please change `logInfo("No op")` to `logInfo("No op and set it to EXECUTE with instants specified.")`

Reply:
the line can not put below `logInfo("No op")`, operator default is  `scheduleAndExecute`, if user specific instants, it need be set to execute after check